File: v2/ruby/lib/articles_api.rb
require "uri"

class ArticlesApi
  basePath = "https://staging.vestorly.com/api/v2"
  # apiInvoker = APIInvoker


  # 
  # Returns all articles
  # @param vestorly_auth Vestorly Auth Token
  # @param limit Limit on the number of articles to return
  # @param text_query Search query parameter
  # @param sort_direction Direction of sort (used with sort_by parameter)
  # @param sort_by Field on model to sort by
  # @return Articles
  def self.findArticles (vestorly_auth, limit, text_query, sort_direction, sort_by, opts={})
    query_param_keys = [:vestorly_auth,:limit,:text_query,:sort_direction,:sort_by]
    headerParams = {}

    
    
    # set default values and merge with input
    options = {
      :'vestorly_auth' => vestorly_auth,
      :'limit' => limit,
      :'text_query' => text_query,
      :'sort_direction' => sort_direction,
      :'sort_by' => sort_by
      
    }.merge(opts)

    #resource path
    path = "/articles".sub('{format}','json')
    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end

    # header parameters
    headers = {}

    _header_accept = ''
    if _header_accept != ''
      headerParams['Accept'] = _header_accept
    end 
    _header_content_type = []
    headerParams['Content-Type'] = _header_content_type.length > 0 ? _header_content_type[0] : 'application/json'

    
    
    # http body (model)
    post_body = nil
    
    # form parameters
    form_parameter_hash = {}
    
    
    response = Swagger::Request.new(:GET, path, {:params=>queryopts,:headers=>headers, :body=>post_body, :form_params => form_parameter_hash }).make.body
     Articles.new(response)
    
    
  
  end

  # 
  # Returns a single article
  # @param id Article Id to fetch
  # @return Articleresponse
  def self.findArticleByID (id, opts={})
    query_param_keys = []
    headerParams = {}

    
    
    # set default values and merge with input
    options = {
      :'id' => id
      
    }.merge(opts)

    #resource path
    path = "/articles/{id}".sub('{format}','json').sub('{' + 'id' + '}', id.to_s)
    
    
    # pull querystring keys from options
    queryopts = options.select do |key,value|
      query_param_keys.include? key
    end

    # header parameters
    headers = {}

    _header_accept = ''
    if _header_accept != ''
      headerParams['Accept'] = _header_accept
    end 
    _header_content_type = []
    headerParams['Content-Type'] = _header_content_type.length > 0 ? _header_content_type[0] : 'application/json'

    
    
    # http body (model)
    post_body = nil
    
    # form parameters
    form_parameter_hash = {}
    
    
    response = Swagger::Request.new(:GET, path, {:params=>queryopts,:headers=>headers, :body=>post_body, :form_params => form_parameter_hash }).make.body
     Articleresponse.new(response)
    
    
  
  end
end

File: v2/csharp/src/main/csharp/vestorly/Api/ArticlesApi.cs
using System;
using System.Collections.Generic;
using RestSharp;
using vestorly.client;
using vestorly.model;

namespace vestorly.api {
  
  public class ArticlesApi {
    string basePath;
    protected RestClient restClient;

    public ArticlesApi(String basePath = "https://staging.vestorly.com/api/v2")
    {
      this.basePath = basePath;
      this.restClient = new RestClient(basePath);
    }

    /// <summary>
    /// Sets the endpoint base url for the services being accessed
    /// </summary>
    /// <param name="basePath"> Base URL
    /// <returns></returns>
    public void SetBasePath(string basePath) {
      this.basePath = basePath;
    }

    /// <summary>
    /// Gets the endpoint base url for the services being accessed
    /// <returns>Base URL</returns>
    /// </summary>
    public String GetBasePath() {
      return this.basePath;
    }

    
    
    /// <summary>
    ///  Returns all articles
    /// </summary>
    /// <param name="vestorlyAuth">Vestorly Auth Token</param>
    /// <param name="limit">Limit on the number of articles to return</param>
    /// <param name="textQuery">Search query parameter</param>
    /// <param name="sortDirection">Direction of sort (used with sort_by parameter)</param>
    /// <param name="sortBy">Field on model to sort by</param>
    /// <returns>Articles</returns>
    public Articles FindArticles (string vestorlyAuth, int? limit, string textQuery, string sortDirection, string sortBy) {

      String myPath = "/articles";

       if ("vestorlyAuth" == "vestorlyAuth") {
        if (("GET" == "POST" || "GET" == "PUT") && "ArticlesApi" != "SessionsApi") {
          myPath += "?vestorly_auth=" + vestorlyAuth;
        }
      } if ("limit" == "vestorlyAuth") {
        if (("GET" == "POST" || "GET" == "PUT") && "ArticlesApi" != "SessionsApi") {
          myPath += "?vestorly_auth=" + limit;
        }
      } if ("textQuery" == "vestorlyAuth") {
        if (("GET" == "POST" || "GET" == "PUT") && "ArticlesApi" != "SessionsApi") {
          myPath += "?vestorly_auth=" + textQuery;
        }
      } if ("sortDirection" == "vestorlyAuth") {
        if (("GET" == "POST" || "GET" == "PUT") && "ArticlesApi" != "SessionsApi") {
          myPath += "?vestorly_auth=" + sortDirection;
        }
      } if ("sortBy" == "vestorlyAuth") {
        if (("GET" == "POST" || "GET" == "PUT") && "ArticlesApi" != "SessionsApi") {
          myPath += "?vestorly_auth=" + sortBy;
        }
      }
      
      var _request = new RestRequest(myPath, Method.GET);


      
      // verify the required parameter 'vestorlyAuth' is set
      if (vestorlyAuth == null) throw new ApiException(400, "Missing required parameter 'vestorlyAuth' when calling FindArticles");
      

      // add default header, if any
      foreach(KeyValuePair<string, string> defaultHeader in ApiInvoker.GetDefaultHeader())
      {
        _request.AddHeader(defaultHeader.Key, defaultHeader.Value);
      }

      _request.AddUrlSegment("format", "json"); // set format to json by default
      
       if (vestorlyAuth != null && ("GET" == "GET" || "GET" == "DELETE" || "ArticlesApi" == "SessionsApi")) _request.AddParameter("vestorly_auth", ApiInvoker.ParameterToString(vestorlyAuth)); // query parameter
       if (limit != null && ("GET" == "GET" || "GET" == "DELETE" || "ArticlesApi" == "SessionsApi")) _request.AddParameter("limit", ApiInvoker.ParameterToString(limit)); // query parameter
       if (textQuery != null && ("GET" == "GET" || "GET" == "DELETE" || "ArticlesApi" == "SessionsApi")) _request.AddParameter("text_query", ApiInvoker.ParameterToString(textQuery)); // query parameter
       if (sortDirection != null && ("GET" == "GET" || "GET" == "DELETE" || "ArticlesApi" == "SessionsApi")) _request.AddParameter("sort_direction", ApiInvoker.ParameterToString(sortDirection)); // query parameter
       if (sortBy != null && ("GET" == "GET" || "GET" == "DELETE" || "ArticlesApi" == "SessionsApi")) _request.AddParameter("sort_by", ApiInvoker.ParameterToString(sortBy)); // query parameter
      
      
      
      

      // make the HTTP request
      IRestResponse response = restClient.Execute(_request);
      if (((int)response.StatusCode) >= 400) {
        throw new ApiException ((int)response.StatusCode, "Error calling FindArticles: " + response.Content);
      }
      return (Articles) ApiInvoker.Deserialize(response.Content, typeof(Articles));
    }
    
    
    /// <summary>
    ///  Returns a single article
    /// </summary>
    /// <param name="vestorlyAuth">Vestorly Auth Token</param>
    /// <param name="id">Article Id to fetch</param>
    /// <returns>Articleresponse</returns>
    public Articleresponse FindArticleByID (string vestorlyAuth, string id) {

      String myPath = "/articles/{id}";

       if ("vestorlyAuth" == "vestorlyAuth") {
        if (("GET" == "POST" || "GET" == "PUT") && "ArticlesApi" != "SessionsApi") {
          myPath += "?vestorly_auth=" + vestorlyAuth;
        }
      }
      
      var _request = new RestRequest(myPath, Method.GET);


      
      // verify the required parameter 'vestorlyAuth' is set
      if (vestorlyAuth == null) throw new ApiException(400, "Missing required parameter 'vestorlyAuth' when calling FindArticleByID");
      
      // verify the required parameter 'id' is set
      if (id == null) throw new ApiException(400, "Missing required parameter 'id' when calling FindArticleByID");
      

      // add default header, if any
      foreach(KeyValuePair<string, string> defaultHeader in ApiInvoker.GetDefaultHeader())
      {
        _request.AddHeader(defaultHeader.Key, defaultHeader.Value);
      }

      _request.AddUrlSegment("format", "json"); // set format to json by default
      _request.AddUrlSegment("id", ApiInvoker.ParameterToString(id)); // path (url segment) parameter
      
       if (vestorlyAuth != null && ("GET" == "GET" || "GET" == "DELETE" || "ArticlesApi" == "SessionsApi")) _request.AddParameter("vestorly_auth", ApiInvoker.ParameterToString(vestorlyAuth)); // query parameter
      
      
      
      

      // make the HTTP request
      IRestResponse response = restClient.Execute(_request);
      if (((int)response.StatusCode) >= 400) {
        throw new ApiException ((int)response.StatusCode, "Error calling FindArticleByID: " + response.Content);
      }
      return (Articleresponse) ApiInvoker.Deserialize(response.Content, typeof(Articleresponse));
    }
    
  }
  
}
